Log responses from Elastic Search

Hi guys,

Really cool open-source project you are running here! Thanks for your work.

I'm having some issues when moving data from MongoDB to ElasticSearch. We have one very big collection (379,000 documents) which may or may not be causing some issues.
When running transporter it seems to move the majority of our documents across to ES but there are a number of collections (mongodb has 406,551 records, only 378,685 in es) missing and transporter exits with the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8bc08e]

goroutine 110 [running]:
panic(0xc36e40, 0xc42000e0e0)
        /usr/local/Cellar/go@1.7/1.7.5/libexec/src/runtime/panic.go:500 +0x1a1
github.com/compose/transporter/pkg/adaptor/elasticsearch/clients/v5.(*Writer).postBulkProcessor(0xc4206f2270, 0x199, 0xc421cc4000, 0x2c, 0x40, 0x0, 0x1263000, 0xc42230e6c0)
        /Users/JP/gocode/src/github.com/compose/transporter/pkg/adaptor/elasticsearch/clients/v5/writer.go:104 +0xfe
github.com/compose/transporter/pkg/adaptor/elasticsearch/clients/v5.(*Writer).(github.com/compose/transporter/pkg/adaptor/elasticsearch/clients/v5.postBulkProcessor)-fm(0x199, 0xc421cc4000, 0x2c, 0x40, 0x0, 0x1263000, 0xc42230e6c0)
        /Users/JP/gocode/src/github.com/compose/transporter/pkg/adaptor/elasticsearch/clients/v5/writer.go:59 +0x73
github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*bulkWorker).commit(0xc420500000, 0x7f2ff414d428, 0xc42000e520, 0x1, 0xc4206380b0)
        /Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk_processor.go:507 +0x360
github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*bulkWorker).work(0xc420500000, 0x7f2ff414d428, 0xc42000e520)
        /Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk_processor.go:443 +0x236
created by github.com/compose/transporter/vendor/gopkg.in/olivere/elastic%2ev5.(*BulkProcessor).Start
        /Users/JP/gocode/src/github.com/compose/transporter/vendor/gopkg.in/olivere/elastic.v5/bulk_processor.go:300 +0x34f

```

When I run transporter and I exclude our large collection, it seems to run okay and gets to the point where it is tailing the MongoDB oplog. However, there is still the issue that not all the data has been moved across. The logs look like this when it is tailing the oplog:
```
INFO[3195] tailing oplog with query map[ts:map[$gte:6394692187568734208]]
  db=main
INFO[3198] Ping for <censored_database_url>:27017 is 17 ms 
INFO[3198] Ping for <censored_database_url> is 20 ms 
INFO[3199] Ping for <censored_database_url> is 24 ms 
INFO[3204] SYNC Starting full topology synchronization... 
INFO[3204] SYNC Processing <censored_database_url>...
INFO[3204] SYNC Processing <censored_database_url>... 
INFO[3204] SYNC Processing <censored_database_url>... 
INFO[3205] SYNC Synchronization was complete (got data from primary). 
INFO[3205] SYNC Synchronization completed: 1 master(s) and 2 slave(s) alive.
```

I was wondering if there is a way to log the response from Elastic Search in order to see if there are any errors? Can this be done using the pipeline?

--- a/adaptor/elasticsearch/clients/v2/writer.go
+++ b/adaptor/elasticsearch/clients/v2/writer.go
@@ -99,12 +99,14 @@ func (w *Writer) Close() {
 }
 
 func (w *Writer) postBulkProcessor(executionID int64, reqs []elastic.BulkableRequest, resp *elastic.BulkResponse, err error) {
-	ctxLog := w.logger.
-		With("executionID", executionID).
-		With("took", fmt.Sprintf("%dms", resp.Took)).
-		With("succeeeded", len(resp.Succeeded()))
+	ctxLog := w.logger.With("executionID", executionID)
+	if resp != nil {
+		ctxLog.With("took", fmt.Sprintf("%dms", resp.Took)).
+			With("succeeeded", len(resp.Succeeded())).
+			With("failed", len(resp.Failed()))
+	}
 	if err != nil {
-		ctxLog.With("failed", len(resp.Failed())).Errorln(err)
+		ctxLog.Errorln(err)
 		return
 	}
 	ctxLog.Infoln("_bulk flush completed")

--- a/adaptor/elasticsearch/clients/v5/writer.go
+++ b/adaptor/elasticsearch/clients/v5/writer.go
@@ -99,12 +99,14 @@ func (w *Writer) Close() {
 }
 
 func (w *Writer) postBulkProcessor(executionID int64, reqs []elastic.BulkableRequest, resp *elastic.BulkResponse, err error) {
-	ctxLog := w.logger.
-		With("executionID", executionID).
-		With("took", fmt.Sprintf("%dms", resp.Took)).
-		With("succeeeded", len(resp.Succeeded()))
+	ctxLog := w.logger.With("executionID", executionID)
+	if resp != nil {
+		ctxLog.With("took", fmt.Sprintf("%dms", resp.Took)).
+			With("succeeeded", len(resp.Succeeded())).
+			With("failed", len(resp.Failed()))
+	}
 	if err != nil {
-		ctxLog.With("failed", len(resp.Failed())).Errorln(err)
+		ctxLog.Errorln(err)
 		return
 	}
 	ctxLog.Infoln("_bulk flush completed")

